compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Dar.hs

```diff
@@ -203,6 +206,32 @@ buildDar service PackageConfigFields {..} ifDir dalfInput = do
                    , Just pkgId
                    )
 
+-- Takes maybe list of dar paths, name version, path
+buildCompositeDar :: [FilePath] -> LF.PackageName -> LF.PackageVersion -> IO (Zip.ZipArchive ())
+buildCompositeDar darPaths name version = do
+  dars <-
+    forM darPaths $ \darPath -> do
+      bs <- BSL.readFile darPath
+      pure $ ZipArchive.toArchive bs
+
+  let darDalfs = fmap (fmap (second3 BSL.toStrict) . either error id . readDalfsWithMeta) dars
+  darLfVersions <- forM darDalfs $ either (fail . DA.Pretty.renderPretty) pure . decodeArchiveLfVersion . snd3 . mainDalf
+
+  let lfVersion =
+        case nubOrd $ LF.versionMajor <$> darLfVersions of
+          -- Note that this will not select dev. We may want to detect if any of the packages are `X.dev` and use this if so (and warn)
```

Comment:
Do we want this?

Reply:
hmm I think here we should just treat X.dev as the upper bound, not as a special case. If any of the included dars are on LF X.dev, that must have been an intentional choice in a nearby `daml.yaml`, or someone is trying to make a composite dar with an X.dev `.dalf` that doesn't otherwise interact with the rest, in which case canton will let them know.

From another point of view, other teams will soon start building projects on 2.dev, so this would prevent them from packaging those projects

Reply:
Agree that .dev should work and be taken as the max - esp. for the dogfooding reasons Moises mentioned.